### PR TITLE
Onboarding all targeted distros to build/test/publish in AMD64 flavor

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

* Onboarding the following distros for build/test/publishing to the `insiders-fast/preview` channels from packages.microsoft.com (Documentation with enlistment instructions will be provided in a follow-up PR)
  * [AlmaLinux 9](https://almalinux.org/)
  * [Amazon Linux 2](https://aws.amazon.com/amazon-linux-2/)
  * [Centos](https://www.centos.org/) 7, 8
  * [Debian](https://www.debian.org/) 12
  * [Red Hat Enterprise Edition](https://www.redhat.com/en/technologies/linux-platforms/enterprise-linux) (RHEL) 7, 8, 9
  * [SUSE Linux Enterprise Server](https://www.suse.com/) 15 (SLES)

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,17 +14,38 @@ jobs:
...
+            { name: mariner-2, arch: amd64 },
+            { name: oraclelinux-7, arch: amd64 },
+            { name: oraclelinux-8, arch: amd64 },
+            { name: rhel-7, arch: amd64 },
+            { name: rhel-8, arch: amd64 },
+            { name: rhel-9, arch: amd64 },
+            { name: sles-15, arch: amd64 },
+            { name: ubuntu-18.04, arch: amd64 },
+            { name: ubuntu-18.04, arch: arm },
+            { name: ubuntu-18.04, arch: arm64 },
```
Comment (MariusNi) [inline .github/workflows/ci.yml:43]: Ubuntu 18 is not in target.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,17 +14,38 @@ jobs:
...
+            { name: ubuntu-20.04, arch: amd64 },
+            { name: ubuntu-20.04, arch: arm },
```
Comment (MariusNi) [inline .github/workflows/ci.yml:45]: >{ name: ubuntu-20.04, arch: arm }, [](http://example.com/codeflow?start=12&length=34)

I believe we can also remove this one (not in use)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -14,17 +14,38 @@ jobs:
...
-            arch: arm64
+        target:
+          [
+            { name: almalinux-9, arch: amd64 },
+            { name: amazonlinux-2, arch: amd64 },
+            { name: centos-7, arch: amd64 },
+            { name: centos-8, arch: amd64 },
+            { name: debian-9, arch: amd64 },
+            { name: debian-9, arch: arm },
+            { name: debian-9, arch: arm64 },
```
Comment (MariusNi) [inline .github/workflows/ci.yml:25]: Not in target. Remove?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -14,17 +14,25 @@ jobs:
...
+            { name: debian-12, arch: amd64 },
+            { name: mariner-2, arch: amd64 },
+            { name: oraclelinux-7, arch: amd64 },
+            { name: oraclelinux-8, arch: amd64 },
+            { name: rhel-7, arch: amd64 },
+            { name: rhel-8, arch: amd64 },
+            { name: rhel-9, arch: amd64 },
+            { name: sles-15, arch: amd64 },
+            { name: ubuntu-20.04, arch: amd64 },
+            { name: ubuntu-22.04, arch: amd64 },
```
Comment (robertschaedler3) [inline .github/workflows/ci.yml:34]: If all the targets are `amd64` we should be able to use container jobs directly which will result in the CI completing faster. 

https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container

**Review by robertschaedler3 — APPROVED** [review]

**Review by MariusNi — APPROVED** [review]
:shipit:

File: .github/workflows/ci.yml (hunk 4 above)
Comment (AhmedBM) [reply .github/workflows/ci.yml:34]: yeah good point, i rather not remove it for now as its likely we will re-introduce

**Pull request merged**